compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEControlCodeToTransaction.cpp

```diff
@@ -63,20 +63,37 @@ LogicalResult convertOp(AMDAIE::NpuWriteBdOp op, TransactionBuilder &builder) {
   return success();
 }
 
+LogicalResult convertOp(AMDAIE::DMAStartOp op, TransactionBuilder &builder) {
+  if (failed(builder.appendDmaStartOp(op))) {
+    return failure();
+  }
+  return success();
+}
+
 LogicalResult controlCodeToTransaction(IRRewriter &rewriter,
                                        AMDAIE::ControlCodeOp controlCodeOp,
                                        TransactionBuilder &builder) {
   SmallVector<Operation *> toBeErased;
-  WalkResult res = controlCodeOp->walk([&](Operation *op) {
+  DenseSet<AMDAIE::LockOp> lockOps;
+  WalkResult res = controlCodeOp->walk([&](AMDAIE::UseLockOp op) {
```

Comment:
This makes sure that all locks are initialized before getting to the other ops? Please add a comment here to explain.